autospec/pkg_integrity.py

```diff
@@ -468,15 +471,20 @@ def verify(self, recursion=False):
             self.print_result(False, err_msg='{} is not a GPG signature'.format(self.package_sign_path))
             os.unlink(self.package_sign_path)
             return None
-        pub_key = self.get_pubkey_path()
+        pub_key = self.get_pub_key(self.pubkey_path)
         EMAIL = parse_key(pub_key, r':user ID packet: ".* <(.+?)>"\n')
-        if not pub_key or os.path.exists(pub_key) is False:
-            key_id = get_keyid(self.package_sign_path)
-            self.print_result(False, 'Public key {} not found in keyring'.format(key_id))
-            if self.interactive is True and recursion is False and attempt_key_import(key_id):
-                print(SEPT)
-                return self.verify(recursion=True)
-            return None
+        if not pub_key or not os.path.exists(pub_key):
```

Comment:
Hrm still seems clunky... Also I don't think EMAIL should be set until after this block since it may not be valid (and since EMAIL is global could potentially be a problem later on).

What about having the pub_key and fallback_pubkey be set in the object, don't even need the get_pub_key function at all then. Gets rid of the extra format for key_fullpath too I think. You can get the keyid as part of the init  since that wouldn't ever change.